build.gradle.kts

```diff
@@ -475,19 +472,30 @@ tasks.rat {
     "**/LICENSE.*",
     "**/NOTICE.*",
     "ROADMAP.md",
-    "clients/client-python/.pytest_cache/*",
-    "clients/client-python/gravitino.egg-info/*",
     "clients/client-python/gravitino/utils/exceptions.py",
     "clients/client-python/gravitino/utils/http_client.py"
   )
 
-  // Add .gitignore excludes to the Apache Rat exclusion list.
-  val gitIgnore = project(":").file(".gitignore")
-  if (gitIgnore.exists()) {
-    val gitIgnoreExcludes = gitIgnore.readLines().filter {
-      it.isNotEmpty() && !it.startsWith("#")
+  // Add .gitignore of subprojects, make them excludes to the Apache Rat exclusion list.
+  val gitIgnores = mutableListOf(
+    project(":").file(".gitignore"),
+    project(":").file("clients/client-python/.gitignore"),
+    project(":").file("web/.gitignore")
+  )
+
+  gitIgnores.forEach { gitIgnore ->
+    if (gitIgnore.exists()) {
+      val path = gitIgnore.relativeTo(projectDir).parent?.let { "$it" } ?: ""
+      gitIgnore.readLines().filter {
+        it.isNotEmpty() && !it.startsWith("#")
+      }.forEach { file ->
+        if (path == "" || file.startsWith("/")) {
+          exclusions.add("$path$file")
+        } else {
+          exclusions.add("$path/$file")
+        }
+      }
     }
```

Comment:
Why not just use "**/.gitignore" in the exclusion list?

Reply:
It's because some folders would include `.gitignore`, like `node_modules`, `htmlcov`, etc. There are additional rules required to avoid them, so I think it's better to list certain `.gitignore` manually.

Reply:
That would only ignore the .gitignore file not folders that contain it.

Reply:
Rat is just a tool that checks for license headers a .gitignore is very unlikely to have a license header so excluding all of them is fine.

Reply:
and even if it does a .giignore file is basically just a list which is not covered by copyright

Reply:
Oh, it's not meant to ignore `.gitignore` file, The `.gitignore` files listed manually (root, client-python and web, here) are read and parsed to add the files and folders list in them to exclusion list, make Rat to ignore them.

For example, there is `venv` in `.gitignore` of client-python project, it will be add in exclusion list so Rat will know that this folder needs not to be checked.

Reply:
That's not the best idea idea just because a file is listed in .gitignore doesn't mean it should be ignored by rat. It also makes it hard to review a release or easily review what we are excluding from an ASF policy/legal point of view. At the ASF Incubator PMC memebers will manually run rat to verify a release contain no unexpcted licensed files.

Reply:
It makes sense, but the original code here works just the same as mine, except it only includes root `.gitignore`, and that's why I think this idea is acceptable.

Reply:
Yep I understod why you did that, but it prbabbly wasn't the best idea to start with for reasons explaned above. I think listing all of the exlustions is probably better so a reviewer can easily see them.